src/app/api/feedback/route.js

```diff
@@ -0,0 +1,21 @@
+import { NextResponse } from 'next/server';
+import { prisma } from '@/lib/db';
+import SendFeedback from '@/lib/validationSchemas/sendFeedbackSchema';
+
+export async function POST(request) {
+  try {
+    const feedbackData = await request.json();
+    const validatedFeedback = SendFeedback.safeParse(feedbackData);
```

Comment:
`import {NextRequest, NextResponse} from 'next/server'

export class BaseError extends Error {
	constructor(message, status = 400, data = null) {
		super(message)
		this.status = status
		this.data = data
	}
}

export const BASE_ERROR_MESSAGES = {
	400: 'Bad Request',
	401: 'Unauthorized',
	402: 'Payment Required',
	403: 'Forbidden',
	404: 'Not found',
	405: 'Method Not Allowed',
	422: 'Unprocessable entity',
	429: 'Too Many Requests',
	500: 'Internal Server Error',
}
export class UnprocessableException extends BaseError {
	constructor(message = BASE_ERROR_MESSAGES[422], data = null) {
		super(message, 422, data)
		this.name = 'Unprocessable entity'
	}
}

export class NotFoundException extends BaseError {
	constructor(message = BASE_ERROR_MESSAGES[404], data = null) {
		super(message, 404, data)
		this.name = 'Not found'
	}
}

export class NoAccessException extends BaseError {
	constructor(message = BASE_ERROR_MESSAGES[403], data = null) {
		super(message, 403, data)
		this.name = 'No Access'
	}
}
export class NotAuthorizedException extends BaseError {
	constructor(message = BASE_ERROR_MESSAGES[401], data = null) {
		super(message, 401, data)
		this.name = BASE_ERROR_MESSAGES[401]
	}
}

export class BadRequestException extends BaseError {
	constructor(message = BASE_ERROR_MESSAGES[400], data = null) {
		super(message, 400, data)
		this.name = BASE_ERROR_MESSAGES[400]
	}
}

export class TooManyRequestsException extends BaseError {
	constructor(message = BASE_ERROR_MESSAGES[429], data = null) {
		super(message, 429, data)
		this.name = BASE_ERROR_MESSAGES[429]
	}
}

export class ServerException extends BaseError {
	constructor(message = BASE_ERROR_MESSAGES[500], data = null) {
		super(message, 500, data)
		this.name = BASE_ERROR_MESSAGES[500]
	}
}

function withErrorHandler(fn) {
	return async function (request, ...args) {
		try {
			return await fn(request, ...args)
		} catch (error) {
			return NextResponse.json(
				{message: error.message || 'Something went wrong'},
				{status: error.status || 500},
			)
		}
	}
}

export const POST = withErrorHandler(async (req: NextRequest) => {
	throw new Error('POST request failed')
})

export const GET = withErrorHandler(async (req: NextRequest) => {
	return NextResponse.json({data: 'GET request was successful'})
})
`


You can handle it this way. Then you won't need to add a try/catch to each route handler